docs/docsite/rst/playbooks_tags.rst

```diff
@@ -90,6 +89,15 @@ All of these apply the specified tags to EACH task inside the play, included
 file, or role, so that these tasks can be selectively run when the playbook
 is invoked with the corresponding tags.
 
+Tags applied to roles will also be applied to all roles listed as a dependency.
+Tags applied to individual tasks within a role *will not* be applied to roles
+listed as dependencies. Tags are inherited *down* the dependency chain. In order
+for tags to be applied to a role and all its dependencies, the tag should be applied
+to the role, not to all the tasks within a role.
```

Comment:
I would actually update the 'dependency' docs indicating that if the dependency is not guarantee of execution as conditionals/tags/etc can decide otherwise. 

Aside from the inheritance, rest seems redundant here.